build-tools/packages/build-cli/src/commands/vnext/check/latestVersions.ts

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { WorkspaceName } from "@fluid-tools/build-infrastructure";
+import { isInternalVersionScheme } from "@fluid-tools/version-tools";
+import * as semver from "semver";
+import { packageOrReleaseGroupArg, semverArg } from "../../../args.js";
+import {
+	BaseCommandWithBuildProject,
+	getAllPackageVersions,
+	sortVersions,
+} from "../../../library/index.js";
+
+type MajorVersion = number;
+
+export default class LatestVersionsCommand extends BaseCommandWithBuildProject<
+	typeof LatestVersionsCommand
+> {
+	static readonly summary =
+		"Determines if an input version matches a latest minor release version. Intended to be used in the Fluid Framework CI pipeline only.";
+
+	static readonly description =
+		"This command is used in CI to determine if a pipeline was triggered by a release branch with the latest minor version of a major version.";
+
+	static readonly args = {
+		version: semverArg({
+			required: true,
+			description:
+				"The version to check. When running in CI, this value corresponds to the pipeline trigger branch.",
+		}),
+		package_or_release_group: packageOrReleaseGroupArg({ required: true }),
+	} as const;
+
+	public async run(): Promise<void> {
+		const { args } = this;
+		const repo = this.getBuildProject();
+		const rgOrPackage = repo.workspaces.get(args.package_or_release_group as WorkspaceName);
```

Comment:
I think you mean release group, not workspace, here. Almost all commands should deal with release groups only. Workspaces are only really relevant if you're changing dependencies, which most commands won't use.

I think we might be missing a function on BuildProject, or a common helper in build-cli. It seems fairly common that we'll want to pass a string to some function and have it return the "matching" packages. In the case of commands, we get this string from either an arg or a flag, and then we need to turn it into a release group. So a utility function that takes a string and does a match on release group name, then if none are found, package name (scoped then unscoped),

I guess what I just described is a version of findPackageOrReleaseGroup that just finds release groups and takes a BuildProject instead of a Context. I.e. findReleaseGroup(rgOrPackageName: string).

I wouldn't go nuts with partial string matching or anything - just match release group names and scoped/unscoped package names.